packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx

```diff
@@ -0,0 +1,435 @@
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  menuShouldBlockScroll?: ReactSelectProps['menuShouldBlockScroll'];
+  /**
+   * Handle change events on the menu input
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  onMenuInputChange?: ReactSelectProps['onInputChange'];
+  /**
+   * Can be used to render a custom value when there are no options (either because of no search results, or all options have been used, or there were none in the first place). Gets called with { inputValue: String }.
+   * <br />
+   * `inputValue` will be an empty string when no search text is present.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  noMenuOptionsMessage?: ReactSelectProps['noOptionsMessage'];
+  /**
+   * Whether to enable search functionality.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  isMenuSearchable?: ReactSelectProps['isSearchable'];
+  /**
+   * Maximum height of the menu before scrolling
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  maxMenuHeight?: ReactSelectProps['maxMenuHeight'];
+  /**
+   * Whether the menu should close after a value is selected. Defaults to `true`.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  closeMenuOnSelect?: ReactSelectProps['closeMenuOnSelect'];
+  /**
+   * Horizontal size limit for the dropdown menu.
+   */
+  menuHorizontalConstraint?: 3 | 4 | 5;
+};
+
+const defaultProps: Pick<
+  TSelectableSearchInputProps,
+  'horizontalConstraint' | 'isClearable' | 'menuHorizontalConstraint'
+> = {
+  horizontalConstraint: 'scale',
+  isClearable: true,
+  menuHorizontalConstraint: 3,
+};
+
+const selectableSearchInputSequentialId = createSequentialId(
+  'selectable-search-input-'
+);
+
+const SelectableSearchInput = (props: TSelectableSearchInputProps) => {
+  const [textInputHasFocus, toggleTextInputHasFocus] = useToggleState(false);
+  const [dropdownHasFocus, toggleDropdownHasFocus] = useToggleState(false);
+  const [searchValue, setSearchValue] = useState(props.value.textValue || '');
+  const containerRef = useRef<HTMLDivElement>(null);
+  const textInputRef = useRef<HTMLInputElement>(null);
+
+  const { isNewTheme } = useTheme();
+
+  const selectablSearchInputId = useFieldId(
+    props.id,
+    selectableSearchInputSequentialId
+  );
+
+  if (!props.isReadOnly) {
+    warning(
+      typeof props.onChange === 'function',
+      'TextInput: `onChange` is required when is not read only.'
+    );
+  }
+
+  const handleTextInputFocus = useCallback(() => {
+    if (props.onFocus)
+      props.onFocus({
+        target: {
+          id: SelectableSearchInput.getTextInputId(selectablSearchInputId),
+          name: getTextInputName(props.name),
+        },
+      });
+    toggleTextInputHasFocus(true);
```

Comment:
I would consider adding brackets to the `if` statements for better readability (not only here)